### PR TITLE
implemented username updating inside the updateUser method

### PR DESCRIPTION
**Pull request by runz0rd**
For issue #357
JIRA JOINDIN-696

### CODE REVIEW

**Comment by liam-wiltshire** [comment]
ok to test

**Comment by liam-wiltshire** [comment]
Thanks for this - I'll try to pull it down and merge it this evening, if not tomorrow

**Comment by runz0rd** [comment]
No problem :+1:

File: src/controllers/UsersController.php [hunk 1]
```diff
@@ -247,6 +247,22 @@ public function updateUser(Request $request, $db)
...
+                if ($existing_user['users']) {
+                    // yes but is that our existing user being found?
+                    $old_user = $user_mapper->getUserById($userId);
+                    if ($old_user['users'][0]['uri'] != $existing_user['users'][0]['uri']) {
+                        // the username exists and not on this user's account
+                        $errors[] = "That username is already associated with another account";
+                    }
+                }
+
+                $user['username'] = filter_var(trim($username), FILTER_SANITIZE_STRING);
```
Comment (heiglandreas) [inline src/controllers/UsersController.php]: Shouldn't that happen right after the ```if (false !== $username) {```?

This is checking an unsanitized username against the database-entries but that username might be changed when sanitized. So IMO the checks should be done **after** sanitizing the username

**Comment by runz0rd** [comment]
You're right. I'll make another pull request with the changes.

**Pull request merged**